probe-rs/src/session.rs

```diff
@@ -117,7 +118,9 @@ impl Session {
         attach_method: AttachMethod,
         permissions: Permissions,
     ) -> Result<Self, Error> {
-        let (mut probe, target) = get_target_from_selector(target, attach_method, probe)?;
+        let registry = Registry::default();
```

Comment:
With the introduction of the registry, maybe it would be good to have a `new_with_registry` variant which allows creating sessions with non-default registries.